### PR TITLE
Add first classes

### PR DESCRIPTION
**Pull request by Samuellekpor**
In this feature branch, we : 

1. Create the class Person and methods
2. Create the class Student inheriting from Person
3. Create the class Teacher inheriting from Person.

### CODE REVIEW

**Review by Ismailco — COMMENTED** [review]
Hi @Samuellekpor, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has some linter errors kindly check your code to fix them and resubmit your project 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Samuellekpor** [comment]
Hi @Ismailco !!!
Thanks for the quick review.
I have been looking for a solution all day. I don't know how to solve the issues and no one can tell me what to do. Can you help me, please?

**Comment by Ismailco** [comment]
Hi @Samuellekpor, I don't know why this is an issue but you need to change the placement of the 'name' parameter in your file:
in the Person the initialize method should be like this: `def initialize(age, name = 'Unknown', parent_permission: true)`
in the Student: `def initialize(classroom, age, name = 'Unknown', parent_permission: true)`
in the Teacher: `def initialize(specialization, age, name = 'Unknown', parent_permission: true)`

and you need to rename the ` is_of_age?` method to ` of_age?`
other errors are just spacing you can fix them with `rubocop -A` or remove and add spaces your self 😄 

if you got any other issues just mention me, and I will do my best to help you
Happy Coding 🚀

**Comment by Samuellekpor** [comment]
Hi @Ismailco !!!
Thanks for the suggestions
I did exactly what you said and it worked. Thanks a lot. I will request another PR right now

**Comment by KDlamini** [comment]
## Status: Invalid 🔴 

### Hi @Samuellekpor 🙋‍♀️,  

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request
 
You have submitted a project with a README that is not documented. Kindly document your README using the new [Microverse template](https://github.com/microverseinc/readme-template).

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_Invalid Code Review Request **does not count** into the code reviews limit._

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+  attr_accessor :name, :age
+
```
Comment (sja-thedude) [inline person.rb]: - [ ] Kindly make sure to add attr_reader and attr_accessor before initialize methods.

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @Samuellekpor, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ All linters passing
✔️ Professional Readme
✔️ Correct Gitflow

## Required 
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Samuellekpor) [reply person.rb]: Hi, @sja-thedude !!!
Thanks for the review. Is there a particular reason for adding the readers and accessors before the initialize method, please?

**Review by codecaiine — APPROVED** [review]
Hi @Samuellekpor , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Professional Readme :heavy_check_mark: 
- Good use Gitflow :heavy_check_mark: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**